src/ReasonReact.re

```diff
@@ -883,6 +883,20 @@ module Router = {
     | Some((window: Dom.window)) =>
       removeEventListener(window, "popstate", watcherID)
     };
+
+  let useUrl = () => {
+    let (url, setUrl) = React.useState(dangerouslyGetInitialUrl);
+
+    React.useLayoutEffect1(
+      () => {
+        let watcherId = watchUrl(url => setUrl(_ => url));
+        Some(() => unwatchUrl(watcherId));
+      },
+      [||],
```

Comment:
`use(Layout)Effect0` should get you this effect already.

Reply:
Ah, right! That's convenient!